plugins/index.js

```diff
@@ -117,156 +182,259 @@ module.exports = (on, config) => {
       } else if (!network) {
         network = 'goerli';
       }
-      const networkChanged = await metamask.changeNetwork(network);
+      const networkChanged = await getProvider(selectedProvider).changeNetwork(
+        network,
+      );
       return networkChanged;
     },
     activateAdvancedGasControlInMetamask: async skipSetup => {
-      const activated = await metamask.activateAdvancedGasControl(skipSetup);
+      const activated = await getProvider(
+        selectedProvider,
+      ).activateAdvancedGasControl(skipSetup);
+      return activated;
+    },
+    activateEnhancedTokenDetectionInMetamask: async skipSetup => {
+      const activated = await getProvider(
+        selectedProvider,
+      ).activateEnhancedTokenDetection(skipSetup);
       return activated;
     },
     activateShowHexDataInMetamask: async skipSetup => {
-      const activated = await metamask.activateShowHexData(skipSetup);
+      const activated = await getProvider(selectedProvider).activateShowHexData(
+        skipSetup,
+      );
       return activated;
     },
     activateTestnetConversionInMetamask: async skipSetup => {
-      const activated = await metamask.activateTestnetConversion(skipSetup);
+      const activated = await getProvider(
+        selectedProvider,
+      ).activateTestnetConversion(skipSetup);
       return activated;
     },
     activateShowTestnetNetworksInMetamask: async skipSetup => {
-      const activated = await metamask.activateShowTestnetNetworks(skipSetup);
+      const activated = await getProvider(
+        selectedProvider,
+      ).activateShowTestnetNetworks(skipSetup);
       return activated;
     },
     activateCustomNonceInMetamask: async skipSetup => {
-      const activated = await metamask.activateCustomNonce(skipSetup);
+      const activated = await getProvider(selectedProvider).activateCustomNonce(
+        skipSetup,
+      );
       return activated;
     },
     activateDismissBackupReminderInMetamask: async skipSetup => {
-      const activated = await metamask.activateDismissBackupReminder(skipSetup);
+      const activated = await getProvider(
+        selectedProvider,
+      ).activateDismissBackupReminder(skipSetup);
       return activated;
     },
-    activateEthSignRequestsInMetamask: async skipSetup => {
-      const activated = await metamask.activateEthSignRequests(skipSetup);
+    activateEnhancedGasFeeUIInMetamask: async skipSetup => {
+      const activated = await getProvider(
+        selectedProvider,
+      ).activateEnhancedGasFeeUI(skipSetup);
       return activated;
     },
-    activateImprovedTokenAllowanceInMetamask: async skipSetup => {
-      const activated = await metamask.activateImprovedTokenAllowance(
-        skipSetup,
-      );
+    activateShowCustomNetworkListInMetamask: async skipSetup => {
+      const activated = await getProvider(
+        selectedProvider,
+      ).activateShowCustomNetworkList(skipSetup);
```

Comment:
What happened here?

Reply:
nothing much, just using providers now

Reply:
I mean that `activateImprovedTokenAllowanceInMetamask` function was replaced by `activateShowCustomNetworkListInMetamask`. Curious what happened here

Reply:
ok, will put that back